src/passes/analysis/verilog.cpp

```diff
@@ -1368,18 +1368,34 @@ Passes::Verilog::compileModuleBody(
         std::move(verilog_connections));
       auto metadata = instance.second->getMetaData();
       if (metadata.count("compile_guard") > 0) {
-        std::vector<std::unique_ptr<vAST::StructuralStatement>> body;
-        body.push_back(std::move(statement));
+        std::vector<std::unique_ptr<vAST::StructuralStatement>> true_body;
+        true_body.push_back(std::move(statement));
         std::string type = metadata["compile_guard"]["type"];
+
+        std::vector<std::unique_ptr<vAST::StructuralStatement>> else_body;
+        RecordType* record_type = cast<RecordType>(instance_module->getType());
+        for (auto field : record_type->getFields()) {
+          Type* field_type = record_type->getRecord().at(field);
+          std::string wire_name = instance.first + "_" + field;
```

Comment:
This seems extremely fragile...

Reply:
This is how output wires are generated by coreir, can you offer a better less fragile solution? The point of having a separate inline pass is that at this point we can assume that the output wires have this name without having to maintain them since the inlining pass handles that.  I'm open to better suggestions but I'm not sure if I see one

Reply:
Read your full review comment after this one, seems like you're just marking place where something could be improved.  We should open an issue to track this and reference the comments here.

Reply:
Yeah, it's probably not worth it to make it more robust for this version. But I think there's a solution that basically solves "all" the problems. My main concern re: fragility here was how "guaranteed" the names for output wires are `instance.first + "_" + field`. E.g. if one line changed somewhere else in how that name was generated, would this break, but the rest of code generation would work (i.e. is not dependent on how those wire names are gen'd)?

Reply:
I think opening a new issue + just a comment here on the explanation of this logic + what it does and doesn't work for would suffice for now (e.g. "handles single driver of output, but not multiple drivers from different conditional blocks etc."